sklearn/decomposition/tests/test_fastica.py

```diff
@@ -238,36 +252,39 @@ def test_fit_transform():
         assert_array_almost_equal(Xt, Xt2)
 
 
-def test_inverse_transform():
+@pytest.mark.filterwarnings("ignore:Ignoring n_components with whiten=False.")
+@pytest.mark.parametrize(
+    "whiten, n_components, expected_shape",
+    [
+        ("arbitrary-variance", 5, (12, 5)),
+        ("arbitrary-variance", 10, (12, 10)),
+        ("unit-variance", 5, (12, 5)),
+        ("unit-variance", 10, (12, 10)),
+        (False, 5, (12, 12)),
+        (False, 10, (12, 12)),
+    ],
+)
+def test_inverse_transform(whiten, n_components, expected_shape):
     # Test FastICA.inverse_transform
-    n_features = 10
     n_samples = 100
-    n1, n2 = 5, 10
     rng = np.random.RandomState(0)
-    X = rng.random_sample((n_samples, n_features))
-    expected = {
-        (True, n1): (n_features, n1),
-        (True, n2): (n_features, n2),
-        (False, n1): (n_features, n2),
-        (False, n2): (n_features, n2),
-    }
-    for whiten in [True, False]:
-        for n_components in [n1, n2]:
-            n_components_ = n_components if n_components is not None else X.shape[1]
-            ica = FastICA(n_components=n_components, random_state=rng, whiten=whiten)
-            with warnings.catch_warnings(record=True):
-                # catch "n_components ignored" warning
-                Xt = ica.fit_transform(X)
-            expected_shape = expected[(whiten, n_components_)]
-            assert ica.mixing_.shape == expected_shape
-            X2 = ica.inverse_transform(Xt)
-            assert X.shape == X2.shape
-
-            # reversibility test in non-reduction case
-            if n_components == X.shape[1]:
-                assert_array_almost_equal(X, X2)
+    X = rng.random_sample((n_samples, 12))
+
+    ica = FastICA(n_components=n_components, random_state=rng, whiten=whiten)
+    Xt = ica.fit_transform(X)
+    assert ica.mixing_.shape == expected_shape
+    X2 = ica.inverse_transform(Xt)
+    assert X.shape == X2.shape
 
+    # reversibility test in non-reduction case
+    if n_components == X.shape[1]:
+        assert_array_almost_equal(X, X2)
```

Comment:
not sure why codecov is complaining here....

Reply:
Neither do I, though it seems not to complain anymore.

Reply:
The shape was changed in the previous commit for `12` instead of `10`, making this condition above false.

This is reverted in dc2e42a272